ql/Cargo.toml

```
@@ -0,0 +1,6 @@
+[workspace]
+members = [
+    "extractor",
+    "generator",
+    "node-types",
```

Have you tried using the components from `ruby` instead. The following might just work and allow us to delete the copied code.
```suggestion
    "../ruby/generator",
    "../ruby/node-types",
```

---

I have not.   
And I would like to not do that for now, and wait until we try to share code between the two extractors.

---

I tried to share them naively, but I run into this issue: https://github.com/rust-lang/cargo/issues/5042  
(I got the `is a member of the wrong workspace` error).   

So it takes a bit more effort than just changing the Cargo.toml.  
I've leave it for now.